creating-virtual-machines/dedicated-cpu.adoc

```diff
@@ -95,6 +95,31 @@ spec:
 [...]
 ----
 
+Requesting dedicated CPU for QEMU emulator
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Apart from VMI's processes, there are additional threads of the QEMU process, such as QEMU main event loop, async I/O operation completion, migration data transfer and more which affect latency and overall performance of the VMIs. In order to enhance the real-time support in KubeVirt, the QEMU emulator threads need to be "isolated" - pinned to a dedicated physical CPU that is not used by any of the VMI virtual CPUs.
+
+This can be defined by specifying `isolateEmulatorThread: true` inside VMI spec's `Spec.Domain.CPU`.
+This will allocate an additional dedicated CPU on the pod exclusively for the emulator thread.
+To use this, `dedicatedCpuPlacement: true` also needs to be there on the VMI spec.
```

Comment:
Thanks, it looks good. However, I think I would write it a bit differently, perhaps something like the following:

A number of QEMU threads, such as QEMU main event loop, async I/O operation completion, etc., also execute on the same physical CPUs as the VMI's vCPUs. This may affect the expected latency of a vCPU.
In order to enhance the real-time support in KubeVirt and provide improved latency, KubeVirt will allocate an additional dedicated CPU, exclusively for the emulator thread, to with it will be pinned. This will effectively "isolate" the emulator thread from the vCPUs of the VMI.

This functionality can be enabled by specifying `isolateEmulatorThread: true` inside VMI spec's `Spec.Domain.CPU` section.
Naturally, this setting has to be specified in a combination with a `dedicatedCpuPlacement: true`.

@stu-gott  what do you think?

Reply:
"to *which* it will be pinned", other than that I like your suggestion @vladikr

Reply:
@vladikr @stu-gott Updated with your suggestions, back to review